src/main/java/org/mule/extension/ftp/internal/operation/FileCommand.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright 2023 Salesforce, Inc. All rights reserved.
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.ftp.internal.operation;
+
+import org.mule.extension.ftp.internal.connection.FileSystem;
+import org.mule.extension.ftp.internal.exception.FileAlreadyExistsException;
+import org.mule.extension.ftp.internal.exception.IllegalPathException;
+import org.slf4j.Logger;
+
+import java.nio.file.Path;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+/**
+ * Extension of {@link AbstractFileCommand} for local file systems which use {@link Path} to identify and manage
+ * files and directories.
+ *
+ * @param <F> the generic type of the {@link FileSystem} on which the operation is performed
+ * @since 1.0
+ */
+public abstract class FileCommand<F extends FileSystem> extends AbstractFileCommand<F, Path> {
+
+  private static final Logger LOGGER = getLogger(FileCommand.class);
+
+  /**
+   * Creates a new instance
+   *
+   * @param fileSystem the {@link FileSystem} on which the operation is performed
+   */
+  protected FileCommand(F fileSystem) {
+    super(fileSystem);
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  protected String pathToString(Path path) {
+    return path.toString();
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @return an absolute {@link Path}
+   */
+  protected Path getAbsolutePath(Path path) {
+    return path.toAbsolutePath();
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @return the parent {@link Path}
+   */
+  protected Path getParent(Path path) {
+    return path.getParent();
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @return the resolved {@link Path}
+   */
+  protected Path resolvePath(Path basePath, String filePath) {
+    return basePath.resolve(filePath);
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @return whether the {@code Path} exists
+   */
+  protected abstract boolean exists(Path path);
+
+  /**
+   * {@inheritDoc}
+   *
+   * @return whether the {@code Path} exists
+   */
+  protected abstract Path getBasePath(FileSystem fileSystem);
+
+  /**
+   * {@inheritDoc}
+   *
+   * @return whether the {@code Path} exists
+   */
+  protected abstract void doMkDirs(Path directoryPath);
+
+  /**
+   * {@inheritDoc}
+   *
+   * @return whether the {@code Path} exists
+   */
+  protected void assureParentFolderExists(Path path, boolean createParentFolder) {
+    super.assureParentFolderExists(path, createParentFolder);
+  }
```

Comment:
This implementation is redundant. Why is this here? Same for all following methods

Reply:
removed